packages/contracts-periphery/contracts/L2/L2ERC721Bridge.sol

```diff
@@ -241,6 +262,39 @@ contract L2ERC721Bridge is Semver, CrossDomainEnabled, Initializable {
         }
     }
 
+    /**
+     * @notice Completes an outbound token transfer. Public function, but can only be called by
+     *         this contract. It's security critical that there be absolutely no way for anyone to
+     *         trigger this function, except by explicit trigger within this contract. Used as a
+     *         simple way to be able to try/catch any type of revert that could occur during an
+     *         ERC721 mint/transfer.
+     *
+     * @param _localToken  Address of the ERC721 on this chain.
+     * @param _remoteToken Address of the corresponding token on the remote chain.
+     * @param _to          Address of the receiver.
+     * @param _tokenId     ID of the token being deposited.
+     */
+    function completeOutboundTransfer(
+        address _localToken,
+        address _remoteToken,
+        address _to,
+        uint256 _tokenId
+    ) external onlySelf {
+        require(
+            // slither-disable-next-line reentrancy-events
+            ERC165Checker.supportsInterface(_localToken, type(IOptimismMintableERC721).interfaceId),
+            "L2ERC721Bridge: local token interface is not compliant"
+        );
+        require(
+            _remoteToken == IOptimismMintableERC721(_localToken).remoteToken(),
+            "L2ERC721Bridge: wrong remote token for Optimism Mintable ERC721 local token"
+        );
+
+        // When a deposit is finalized, we give the NFT with the same tokenId to the account
+        // on L2.
+        IOptimismMintableERC721(_localToken).safeMint(_to, _tokenId);
```

Comment:
Note that this introduces the possibility of a callback from the `_to` address. AFAICT that's OK because: 

2. This method is `onlySelf`
3. Thus the only way to call this method is via `finalizeBridgeERC721` which has the `onlyFromCrossDomainAccount` modifier, meaning it can only be called by the xDM. 
4. The xDM's relayMessage method is [nonReentrant](https://github.com/ethereum-optimism/optimism/blob/111bdcdaf4c7ac7659936c1c78aaad3c6b7b9c91/packages/contracts/contracts/L1/messaging/L1CrossDomainMessenger.sol#L171).